code/datums/gamemode/role/syndicate.dm

```diff
@@ -197,4 +197,30 @@
 	logo_state = "nuke-logo"
 
 /datum/role/nuclear_operative/leader
+	name = NUKE_OP_LEADER
+	id = NUKE_OP_LEADER
+	required_pref = NUKE_OP
+	disallow_job = TRUE
 	logo_state = "nuke-logo-leader"
+
+/datum/role/nuclear_operative/leader/OnPostSetup()
+	if(antag && src.faction.stage != FACTION_ENDGAME)
+		var/datum/action/play_ops_music/go_loud = new /datum/action/play_ops_music(antag)
+		go_loud.Grant(antag.current)
+	..()
+	
+/datum/action/play_ops_music
+	name = "Go Loud"
+	desc = "For the operative who prefers style over subtlety."
+	icon_icon = 'icons/obj/device.dmi'
+	button_icon_state = "megaphone"
+
+/datum/action/play_ops_music/Trigger()
+	var/mob/living/M = owner
+	var/datum/faction/F
+	var/confirm = alert(M, "Are you sure you want to announce your presence? Doing so will display a command announcement and start the Nuclear Assault playlist.", "Are you sure?", "No", "Yes")
+	if (confirm == "Yes" && M.stat == CONSCIOUS)
```

Comment:
ok but I'm concerned you can spam click on the action button and then click yes on a bunch of prompts to spam the alert, that's why I told you to check if the stage was already ENDGAME in the earlier review